modules/imgproc/src/featureselect.cpp

```diff
@@ -305,15 +299,36 @@ void cv::goodFeaturesToTrack( InputArray _image, OutputArray _corners,
         for( int x = 1; x < imgsize.width - 1; x++ )
         {
             float val = eig_data[x];
-            if( val != 0 && val == tmp_data[x] && (!mask_data || mask_data[x]) )
+            if( val != 0 && (fabs(val - tmp_data[x])<CV_GFTTEPS) && (!mask_data || mask_data[x]) )
                 tmpCorners.push_back(eig_data + x);
         }
     }
-    std::sort( tmpCorners.begin(), tmpCorners.end(), greaterThanPtr() );
 
     std::vector<Point2f> corners;
     size_t i, j, total = tmpCorners.size(), ncorners = 0;
 
+    if (total == 0)
+    {
+        _corners.release();
+        return;
+    }
+
+// sort tmpCorners
+
+    float* pTmp;
+    float ** vec = (float **) &tmpCorners[0];
+
+    for(i=0;i<total-1;i++)
+    {
+       for(j=i+1;j<total;j++)
+       {
+           if(*vec[j] > *vec[i])
+           {
+               pTmp = vec[i]; vec[i] = vec[j];vec[j] = pTmp;
+           }
+       }
+    }
```

Comment:
Hello!

I have finished working with code for this (#3540) pull request. All test passed.
Could you review my code?

Thanks,
Alexander


From: Alexander Alekhin [mailto:notifications@github.com]
Sent: Monday, August 10, 2015 4:26 PM
To: Itseez/opencv
Cc: Usentsov, Alexander
Subject: Re: [opencv] added test goodfeaturetotrack.cpp (#3540)


In modules/imgproc/src/featureselect.cpp<https://github.com/Itseez/opencv/pull/3540#discussion_r36629704>:

> +

> +// sort tmpCorners

> +

> +    float* pTmp;

> +    float ** vec = (float **) &tmpCorners[0];

> +

> +    for(i=0;i<total-1;i++)

> +    {

> +       for(j=i+1;j<total;j++)

> +       {

> +           if(*vec[j] > *vec[i])

> +           {

> +               pTmp = vec[i]; vec[i] = vec[j];vec[j] = pTmp;

> +           }

> +       }

> +    }

std::sort above should work fine. Revert this change.

—
Reply to this email directly or view it on GitHub<https://github.com/Itseez/opencv/pull/3540/files#r36629704>.

--------------------------------------------------------------------
Joint Stock Company Intel A/O
Registered legal address: Krylatsky Hills Business Park,
17 Krylatskaya Str., Bldg 4, Moscow 121614,
Russian Federation

This e-mail and any attachments may contain confidential material for
the sole use of the intended recipient(s). Any review or distribution
by others is strictly prohibited. If you are not the intended
recipient, please contact the sender and delete all copies.